### DISCUSSION TITLE
XInput - missing constant? ERROR_SUCCESS

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [microsoft/CsWin32#618] [Q&A] [answered]
I'm setting up XInput. And I should see a constant ERROR_SUCCESS. I have the following using...
```
using Windows.Win32.UI.Input.XboxController;
```
And I've also put this line in NativeMethods...
```
ERROR_SUCCESS
```
Is it missing or is the normal usage replaced with something else?

Edit: I've read somewhere that ERROR_SUCCESS = 0, where are these values defined?

**Comment by AArnott** [comment, marked as answer]
With `ERROR_SUCCESS` a warning is produced by the build:

> 1>C:\Users\andarno\source\repos\CsWin32Sandbox\CsWin32Sandbox\NativeMethods.txt(1,1,1,14): warning PInvoke004: Use the name of the enum that declares this constant: WIN32_ERROR

The constant is still emitted, but it's as an enum value within the `WIN32_ERROR` enum. That's where you'll find it.

![image](https://user-images.githubusercontent.com/3548/180609373-0b166039-41d9-440e-a817-4a2973f2ac34.png)

**Reply by Gavin-Williams** [reply]
I had warnings disabled. Thank you for bringing that to my attention!
